x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_service/index.ts

```diff
@@ -229,29 +245,124 @@ export class AIAssistantService {
         pluginStop$: this.options.pluginStop$,
       });
 
-      await this.knowledgeBaseDataStream.install({
-        esClient,
-        logger: this.options.logger,
-        pluginStop$: this.options.pluginStop$,
-      });
+      if (this.assistantDefaultInferenceEndpoint) {
```

Comment:
This if condition looks long enough to potentially going into a separate funtion.